linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/ResponseClassifiers.scala

```diff
@@ -58,14 +60,32 @@ object ResponseClassifiers {
     }
   }
 
+  object RetryableException {
+    def unapply(e: Throwable): Boolean = e match {
+      case RetryableWriteException(_) => true
+      case Failure(Some(_: TimeoutException)) => true
+      case Failure(Some(_: UtilTimeoutException)) => true
+      case _: TimeoutException => true
+      case _: UtilTimeoutException => true
+      case _: ChannelClosedException => true
+      case _ => false
+    }
```

Comment:
retry policies take a `Try[Nothing]` and we are handling a `Try[Any]` (from ReqRep), so I'm not sure they can be reused here:

```
[error] /Users/ver/b/l5d/linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/ResponseClassifiers.scala:77: type mismatch;
[error]  found   : PartialFunction[com.twitter.util.Try[Nothing],Boolean]
[error]  required: PartialFunction[com.twitter.util.Try[Any],Boolean]
[error]       TimeoutAndWriteExceptionsOnly.orElse(ChannelClosedExceptionsOnly)
[error]   
```